### PR TITLE
New preprocess methods

### PR DESCRIPTION
**Pull request by msilvafe**
Adding new preprocess methods for testing full CMB preprocess configuration and some error handling in the `preprocess_tod` site pipeline script.

### CODE REVIEW

File: sotodlib/preprocess/processes.py [hunk 1]
```diff
@@ -730,6 +730,85 @@ def plot(self, aman, proc_aman, filename):
...
+
+    def process(self, aman, proc_aman):
+        _f = getattr(tod_ops.filters,
+                self.process_cfgs.get('filt_function','high_pass_butter4'))
+        filt = _f(**self.process_cfgs.get('filter_params'))
+        aman[self.wrap_name] = tod_ops.filters.fourier_filter(aman, filt,
+                                        signal_name=self.signal_name)
+        if self.process_cfgs.get("trim_samps"):
+            trim = self.process_cfgs["trim_samps"]
+            aman.restrict('samps',(trim, -trim))
```
Comment (tterasaki) [inline sotodlib/preprocess/processes.py:766]: I think
```
aman.restrict('samps',(aman.samps.offset + trim, aman.samps.offset + aman.samps.count - trim))
```
is better.

**Comment by tterasaki** [comment]
I really want to merge this into master as my pointing workflow is branched from this and using preprocess functions.
Do you think it is ready to go?

**Comment by msilvafe** [comment]
> I really want to merge this into master as my pointing workflow is branched from this and using preprocess functions. Do you think it is ready to go?

Hey @tterasaki, I think it still needs a little work but I didn't know that you were using it and it was holding up the pointing workflow work so thanks for letting me know. I will focus on getting this ready for review.

**Comment by msilvafe** [comment]
Ok, I've tested this and updated it accordingly. I think it's ready now.

**Review by tterasaki — APPROVED** [review]
Sorry, I have missed it. Approving.

**Pull request merged**